cmd/balance-agent/main.go

```diff
@@ -508,6 +515,54 @@ func mkGetLndApi(ctx *cli.Context) agent_entities.NewApiCall {
 	}
 }
 
+func reloadConfig(ctx context.Context, f *filter.FileFilter) {
+	glog.Info("Reloading configuration...")
+
+	if f != nil {
+		err := f.Reload()
+		if err != nil {
+			glog.Warningf("Error while reloading configuration %v", err)
+		}
+	}
+}
+
+func signalHandler(ctx context.Context, f filter.FilterInterface) {
+	ff, ok := f.(*filter.FileFilter)
+
+	signal_chan := make(chan os.Signal, 1)
+	exit_chan := make(chan int)
+
+	signal.Notify(signal_chan,
+		syscall.SIGHUP)
+	go func() {
+		for {
+			select {
+			case s := <-signal_chan:
+				switch s {
+				case syscall.SIGHUP:
+					if ok {
+						reloadConfig(ctx, ff)
+					} else {
+						reloadConfig(ctx, nil)
+					}
+
+				case syscall.SIGTERM:
+					exit_chan <- 0
+				case syscall.SIGQUIT:
+					exit_chan <- 0
+				default:
+					fmt.Println("Unknown signal.")
+					exit_chan <- 1
+				}
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+	code := <-exit_chan
+	os.Exit(code)
```

Comment:
For long term plan I dont like this. When we will have multiple go routines this will be problematic for gracefull shutdown.

I think go routines should return error and in `main()` we wait for error so we can initiate gracefull shutdown for other routines. So signal handling shoud be directly in main()